tests/components/lupusec/test_config_flow.py

```diff
@@ -0,0 +1,260 @@
+    ) as mock_step_user:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_DATA_STEP,
+        )
+    await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+    assert len(mock_step_user.mock_calls) == 1
+
+
+@pytest.mark.parametrize(
+    ("raise_error", "text_error"),
+    [
+        (LupusecException("Test lupusec exception"), "cannot_connect"),
+        (Exception("Test unknown exception"), "unknown"),
+    ],
+)
+async def test_flow_user_init_data_error_and_recover(
+    hass: HomeAssistant, raise_error, text_error
+) -> None:
+    """Test exceptions and recovery."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "lupupy.Lupusec.__init__",
+        side_effect=raise_error,
+    ) as mock_initialize_lupusec:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_DATA_STEP,
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == FlowResultType.FORM
+        assert result2["errors"] == {"base": text_error}
+
+        assert len(mock_initialize_lupusec.mock_calls) == 1
+
+        # Recover
+        mock_initialize_lupusec.side_effect = None
+        mock_initialize_lupusec.return_value = None
+        with patch(
+            "homeassistant.components.lupusec.async_setup_entry",
+            return_value=True,
+        ) as mock_setup_entry:
+            result3 = await hass.config_entries.flow.async_configure(
+                result["flow_id"],
+                MOCK_DATA_STEP,
+            )
+
+            await hass.async_block_till_done()
+
+            assert result3["type"] == FlowResultType.CREATE_ENTRY
+            assert result3["title"] == MOCK_DATA_STEP[CONF_HOST]
+            assert result3["data"] == MOCK_DATA_STEP
+            assert len(mock_setup_entry.mock_calls) == 1
+            assert len(mock_initialize_lupusec.mock_calls) == 2
+
+
+async def test_flow_user_init_data_already_configured(hass: HomeAssistant) -> None:
+    """Test duplicate config entry.."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title=MOCK_DATA_STEP[CONF_HOST],
+        data=MOCK_DATA_STEP,
+        unique_id=MOCK_DATA_STEP[CONF_HOST],
+    )
+
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.lupusec.config_flow.test_host_connection",
+        return_value=None,
+    ) as mock_test_host_connection:
```

Comment:
Oh right, I see something we should change in the config flow -> (see next comment)

Reply:
done.